aether-ui/aether/ui/api/serializers.py

```diff
@@ -47,6 +47,24 @@ class PipelineSerializer(DynamicFieldsMixin, serializers.ModelSerializer):
 
     contracts = ContractSerializer(many=True, read_only=True)
 
+    def update(self, instance, validated_data):
+        read_only_contracts = instance.contracts.filter(is_read_only=True)
+        if read_only_contracts:
+            raise serializers.ValidationError({
+                'description': 'Input is readonly'
```

Comment:
As an enhancement, could you start using the translation library for the messages in the python side?

```python
from django.utils.translation import ugettext as _

                # ...
                'description': _('Input is readonly')
```